### ISSUE TITLE
Add .net core build and nuget package

### ISSUE THREAD

**Issue by ajbeaven** [jamesmanning/RunProcessAsTask#7]
```
error: Package RunProcessAsTask 1.1.3 is not compatible with netcoreapp1.0
```

Thanks for the repo! Very helpful :)

**Comment by jamesmanning** [comment]
@ajbeaven FYI, I'm looking around and trying to figure out how to add support for .net core now - certainly nothing in the current code is Windows-specific AFAIK (doesn't use shell execute, for instance), but I haven't targeted .net core before.

**Comment by ajbeaven** [comment]
Legend, thanks for that! Yes, I've currently just copy pasted the source code and everything but the securitystring was supported it seemed. Fortunately I didn't need that so stripped it out.

**Comment by jamesmanning** [comment]
@ajbeaven nice!  I'll stick that overload in an appropriate #if once I can figure out the rest of it. 😄

**Comment by jamesmanning** [comment]
After looking into this more and starting down the path of getting a .net core build working (with a functional project.json and a new csproj that target .netstandard 1.6), since there's a pretty simple current workaround, I'm going to hold off on having this target .net core until VS15 goes RTM since my understanding of the tooling is that at that point I should be able to fully target .netstandard 1.6 from just the csproj and won't need to create/maintain project.json (which is going away).

The VS 2015 tooling isn't horrible, but there's a bit too much hackery with the tooling as it stands and I'd rather use the better tooling and target .netstandard the Right Way since what I've got building now seems to work, but I'm a bit fearful would do more harm than good 

- need to go edit project properties after creation to change from targeting portable to .net standard, then need to change version to 1.6 https://docs.microsoft.com/en-us/dotnet/articles/core/tutorials/target-dotnetcore-with-msbuild
- need `<NuGetTargetMoniker>.NETStandard,Version=v1.6</NuGetTargetMoniker>` added near end of csproj https://github.com/dotnet/roslyn/issues/12918

Hopefully VS15 RTM will happen soon enough - it just feels like at the moment it's a lot more work and duct tape to get this up and working than it will be once VS15 is out and if I can avoid having to adopt project.json just to turn around and abandon it, I'd rather just wait a little. 😄

**Comment by ajbeaven** [comment]
Totally understand and really appreciate you taking the time to look in to it.

**Comment by jamesmanning** [comment]
@ajbeaven since my earlier effort to hold off on this was until Visual Studio 2017 was RTM and that has now happened, I'm revisiting this and hoping I can get a .NETStandard 1.6 build going. 😄

**Comment by ajbeaven** [comment]
Great, thanks James!

**Comment by MaxCheetham** [comment]
@jamesmanning If it helps with the exeption to the use of SecureString the entire codebase is compatable with .net 1.x.

**Comment by bbrandt** [comment]
@MaxCheetham 
Seems like SecureString is available in .NET Standard based on this:
https://github.com/dotnet/corefx/issues/1387
https://www.nuget.org/packages/System.Security.SecureString/

But, not available in .NET Standard based on the docs:
https://docs.microsoft.com/en-us/dotnet/api/system.security.securestring?view=netcore-1.1#remarks

Have you tested it or have additional info?

**Comment by MaxCheetham** [comment]
Interesting, I can test after work tonight?

**Comment by jamesmanning** [comment]
@bbrandt looking at [their implementation linked from the issue](https://github.com/dotnet/coreclr/blob/e74cdcb1ed6021eaf03eea5ee7f6ba3c6b403daf/src/mscorlib/corefx/System/Security/SecureString.Unix.cs), no actual encryption seems to take place, so at the moment I'm much more likely to just kill off that overload since it seems to be more trouble than it's worth, and the (hopefully few) people that really need a SecureString version can always just use Process directly.  Does that sound ok?

@MaxCheetham that seem ok?

**Comment by MaxCheetham** [comment]
@jamesmanning yeah that seems fine, people can always write there own extension method if needed. Is there anything you need a hand with?

**Comment by bbrandt** [comment]
@jamesmanning I'm good with that decision.

**Comment by jamesmanning** [comment]
@MaxCheetham honestly? yeah, a PR to modify/recreate this as a .net core project would be wonderful 😄 AFAICT the resulting nuget package would only need to target .net standard 1.0 and not multiple targets (keeping around a net46 target for instance) but I haven't actually verified that. I'm happy that NuGet 4.x now lets us do away with the nuspec too. Depending on VS2017 would be fine with me if that makes it easier. 😄

**Comment by MaxCheetham** [comment]
@jamesmanning Sure I can do that. i'll look into NuGet for Core, any pointers?

**Comment by jamesmanning** [comment]
@MaxCheetham i just have the link I had started with @ https://docs.microsoft.com/en-us/dotnet/articles/core/deploying/creating-nuget-packages

**Comment by MaxCheetham** [comment]
@jamesmanning What branch do you want the pr aimed at?

**Comment by jamesmanning** [comment]
@MaxCheetham master's fine with me

**Comment by MaxCheetham** [comment]
@jamesmanning Any chance you know how to force appvoyer to use msbuild version '15.0'?

**Comment by bbrandt** [comment]
Use the Visual Studio 2017 image?  Maybe?

https://github.com/appveyor/ci/issues/1179
https://github.com/lexicalunit/nanodbc/issues/247

```
environment:
  matrix:
  - platform: x64
    APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
  - platform: x64
    APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
```

**Comment by MaxCheetham** [comment]
@jamesmanning  #12 is locally passing the test suit whilst running netcoreapp1.1. I'm pretty sure that the integration tests on here are failing due to their config. Can you offer any advice?

**Comment by jamesmanning** [comment]
@MaxCheetham bear with me since I'm still quite new to all this.

TBH what I expected to have happen was that the project would change to be netstandard1.0 instead of netcoreapp since I think that means it can continue being consumed by .NET (at least 4.6?) but also be consumable by .NET Core.  My understanding (which may be wrong!) is netcoreapp would limit its usefulness and we'd be stuck going down the path of multiple builds for different frameworks, which I was hoping to avoid since they did all that work to create the .NET Standard and I'm _hoping_ that means we can avoid the multiple-targets route.

Ignoring that for the moment, if things are passing for you locally and the issue is just the CI hosts not supporting .NET Core at the moment, I wouldn't worry about continuing to bang on it.

**Comment by MaxCheetham** [comment]
@jamesmanning righto. I'll give the .yml's a look into over the weekend and try and get to the point where we are all green on integration tests. Sorry about the .NET Core/ netstandard1.0 mixup. I'm more than happy to look into that over the weekend.

What would your suggestions with the pr be moving forward?

**Comment by jamesmanning** [comment]
@MaxCheetham if netstandard1.0 is possible (which API-wise I believe it is), then that'd be a great target for the weekend.

The use case I'm most worried about is one I *think* should work but I'm not 100% sure.

- User is currently using the existing nuget package with a .NET Framework app/library
- User upgrades to new version which only targets netstandard1.0
- Everything Just Works for them (this might require that they're using 4.6 for netstandard1.0 to work, but I'm not overly worried about that at the moment)

**Comment by jamesmanning** [comment]
@MaxCheetham as long as the 'existing version nuget consumer' scenario is working (and, of course, '.net core nuget consumer' since that's what we're enabling 😄  ) and the tests pass locally, I'm not as worried about the CI's failing.  It's great if you can get the CI's working as well, but that's a lower priority IMHO than the nuget package use cases, especially since the CI setups are, at least currently, just testing the code and not a built nuget package, which is what I'm more worried about breaking at the moment. 😄

**Comment by MaxCheetham** [comment]
@jamesmanning unfortunently System.Diagnostics.Process has a requierment of .NETStandard, Version=v1.3 #c72d266 is now supportive of netstandard1.4. So if you want to package that up then the [tests](https://i.gyazo.com/08ad5655f4a1a789c548782c2763a9d4.png) are green on my local.

**Comment by EamonNerbonne** [comment]
So, I'd recommend merging soon enough and doing some post-work.

There's a few things to consider, which don't all need to wait.

Hand-editing: First of all, it's more normal now (and easier) to edit the project files by hand.  You're going to want to be able to do that because the UI isn't always very good; sometimes it's downright misleading (it's clearly still playing catch-up).  And to do that, it's important for the project file to have a limited amount of "gunk".  The default project files the old tooling generated had lots of gunk; most you can simply throw away.  #12 is already pretty slimmed down, but I'd also get rid of all the conditional stuff you can.  You need very little.

At least consider multi-targetting: Sure, in theory you can target .net standard 1.x and support both .net core and .net framework, but in practice, that provides a pretty poor .net framework experience for two reasons.
 - Firstly, for some reason I don't fully understand, .net standard support has a high .net framework requirement.  even [v1.3 means a minimum .net framework of 4.6](https://blogs.msdn.microsoft.com/dotnet/2016/09/26/introducing-net-standard/), and if you want  1.6 (the highest version supported by .net core 1.0, which is still a considerably smaller api than the full framework), then you've raised the bar to [.net framework 4.7](https://blogs.msdn.microsoft.com/dotnet/2017/04/05/announcing-the-net-framework-4-7/).  Basic reflection support [was added in 1.5](https://github.com/dotnet/standard/blob/master/docs/versions/netstandard1.5_diff.md), so since you can avoid that dependency, this applies less to RunProcessAsTask.
 - Secondly, if you reference a .net standard lib from a .net framework project, that means adding lots of dependencies on "core system" assemblies from the .net standard stack.  When I updated a very simple project I maintain to be based on .net standard 1.1, somebody noted that they weren't happy with seeing lots of extra dependencies in nuget, and needing 22 extra dlls on disk: https://github.com/EamonNerbonne/a-vs-an/issues/2  Enabling multitargeting can be really easy: https://github.com/EamonNerbonne/a-vs-an/commit/fa1e67a5eede46af0b0fe3f0bef1e0097ca0088d . Now, I think that lots and lots of dependencies is likely at least partially a sign of the times that you're not entirely going to be able to avoid, but I minimizing dependencies is still an entirely reasonable goal.  It's a bit of a pain that lib authors need to take steps to multitarget, but it's not a huge deal (well, except for testing, I have no idea how to solve that yet).

**Comment by MaxCheetham** [comment]
@EamonNerbonne You are correct when it comes to the dependencies everywhere. @jamesmanning Am i correct in the asumption that only the RunProcessAsTask solution that needs to be migrated. You don't pack the test suit?

**Comment by jamesmanning** [comment]
@MaxCheetham correct, only the main project gets packaged.  The test suite is just run via the built dll, no packaging needed/done.

**Comment by EamonNerbonne** [comment]
The issue with testing that I meant is that if you multitarget, then really should should be testing every target, especially if you're doing anything that might plausibly be target dependent.

**Comment by MaxCheetham** [comment]
.NET Core Now Supported (at least on Windows) @ajbeaven #14

**Comment by ajbeaven** [comment]
Thanks guys! Looking forward to the nuget package being updated.

**Comment by jamesmanning** [comment]
@ajbeaven I've uploaded the 1.2.0 nuget package so please give it a shot when you have a chance.

@EamonNerbonne @MaxCheetham changes worked great, I just did msbuild /t:pack /p:Configuration=Release then nuget push - great job guys! 👍
